addons/account/wizard/account_payment_register.py

```diff
@@ -148,17 +149,70 @@ def _get_batches(self):
         if not lines:
             raise UserError(_("You can't open the register payment wizard without at least one receivable/payable line."))
 
+        # Lines are grouped first on 'partner_id','account_id','currency_id' & 'partner_type' and
+        # then on 'partner_bank_id' & 'payment_type'. 'batches' data structure :
+        # {common_pay_val_key:{
+        #         part_bank_pay_type_key : lines,
+        #         part_bank_pay_type_key : lines,
+        #  }}
         batches = {}
+
         for line in lines:
             batch_key = self._get_line_batch_key(line)
-
-            serialized_key = '-'.join(str(v) for v in batch_key.values())
-            batches.setdefault(serialized_key, {
-                'key_values': batch_key,
-                'lines': self.env['account.move.line'],
+            common_pay_val_key = frozendict(batch_key)
+            part_bank_pay_type_key = frozendict({
+                'payment_type': 'inbound' if line.balance > 0.0 else 'outbound',
+                'partner_bank_id': line.move_id.partner_bank_id.id,
```

Comment:
Don't create a dict and cast it to a frozendict, create the fronzendict with the kwargs.